azext_iot/iothub/edge_device_config.py

```diff
@@ -0,0 +1,469 @@
+
+EDGE_CONFIG_SCRIPT_HEADERS = """
+# This script will attempt to configure a pre-installed iotedge as a nested node.
+# It must be run as sudo, and will modify the ca
+
+device_id="{}"
+cp config.toml /etc/aziot/config.toml
+"""
+EDGE_CONFIG_SCRIPT_HOSTNAME = """
+# ======================= Set Hostname =======================================
+
+read -p "Enter the hostname to use: " hostname
+if [ -z "$hostname" ]
+then
+    echo "Invalid hostname $hostname"
+    exit 1
+fi
+
+sed -i "s/{{HOSTNAME}}/$hostname/" /etc/aziot/config.toml
+"""
+EDGE_CONFIG_SCRIPT_PARENT_HOSTNAME = """
+# ======================= Set Parent Hostname =======================================
+
+read -p "Enter the parent hostname to use: " parent_hostname
+if [ -z "$parent_hostname" ]
+then
+    echo "Invalid parent hostname $parent_hostname"
+    exit 1
+fi
+
+sed -i "s/{{PARENT_HOSTNAME}}/$parent_hostname/" /etc/aziot/config.toml
+"""
+EDGE_CONFIG_SCRIPT_CA_CERTS = f"""
+# ======================= Install nested root CA =======================================
+if [ -f /etc/os-release ]
+then
+        . /etc/os-release
+        if [[ "$NAME" == "Common Base Linux Mariner"* ]];
+        then
+                cp {EDGE_ROOT_CERTIFICATE_FILENAME} /etc/pki/ca-trust/source/anchors/{EDGE_ROOT_CERTIFICATE_FILENAME}.crt
+                update-ca-trust
+        else
+                cp {EDGE_ROOT_CERTIFICATE_FILENAME} /usr/local/share/ca-certificates/{EDGE_ROOT_CERTIFICATE_FILENAME}.crt
+                update-ca-certificates
+        fi
+else
+        cp {EDGE_ROOT_CERTIFICATE_FILENAME} /usr/local/share/ca-certificates/{EDGE_ROOT_CERTIFICATE_FILENAME}.crt
+        update-ca-certificates
+fi
+
+systemctl restart docker
+
+# ======================= Copy device certs  =======================================
+cert_dir="/etc/aziot/certificates"
+mkdir -p $cert_dir
+cp "{EDGE_ROOT_CERTIFICATE_FILENAME}" "$cert_dir/{EDGE_ROOT_CERTIFICATE_FILENAME}"
+cp "$device_id.full-chain.cert.pem" "$cert_dir/$device_id.full-chain.cert.pem"
+cp "$device_id.key.pem" "$cert_dir/$device_id.key.pem"
+"""
+EDGE_CONFIG_SCRIPT_HUB_AUTH_CERTS = """
+# ======================= Copy hub auth certs  =======================================
+cert_dir="/etc/aziot/certificates"
+mkdir -p $cert_dir
+cp "$device_id.hub-auth-cert.pem" "$cert_dir/$device_id.hub-auth-cert.pem"
+cp "$device_id.hub-auth-key.pem" "$cert_dir/$device_id.hub-auth-key.pem"
+"""
+EDGE_CONFIG_SCRIPT_APPLY = """
+# ======================= Read User Input =======================================
+iotedge config apply -c /etc/aziot/config.toml
+
+echo "To check the edge runtime status, run 'iotedge system status'. To validate the configuration, run 'sudo iotedge check'"
+"""
+
+EDGE_SUPPORTED_OS_LINK = (
+    "https://docs.microsoft.com/en-us/azure/iot-edge/support?view=iotedge-2020-11"
+)
+
+EDGE_LINUX_TUTORIAL_LINK = (
+    "https://learn.microsoft.com/en-us/azure/iot-edge/how-to-provision-single-device-linux-symmetric"
```

Comment:
I wonder if these links should be aka.ms ones, unless they will stay the same forever.